### DISCUSSION TITLE
Create Pull Requests on Mobile

### DISCUSSION THREAD

**Discussion by justice-sh** [community/community#6348] [Mobile] [answered]
Could the feature for creating pull requests on mobile phones be implemented?

**Comment by brianlovin** [comment]
Thanks for the note @Sherlock-HolmesJM — are you envision that this feature would include file editing and committing from your phone? It would be helpful for us if you could share a few more details about what kind of flow you're in where opening a pull request on mobile would be helpful.

Thanks!

**Reply by justice-sh** [reply]
Thank you for responding.

The workflow is simple: the PR we make on the browser, would be nice to do on the mobile app as well. Just a replica of what already is.

**Reply by brianlovin** [reply]
In the browser, you can make a pull request for recently pushed branches, but you can also edit files directly in the web browser and commit over time. Are you envisioning one of these flows to be more useful for you?

1. You've pushed a branch to your repo, but haven't created a pull request yet. You want the ability to open a pull request for that branch.
2. You want to edit files on the mobile app, and have those edits appear as commits in a new pull request.

**Reply by justice-sh** [reply]
1. Yes, I want the ability to open a pull request with the GitHub Mobile App.

2. Not really, I don't care about editing the files. I just want to make a PR from the GitHub Mobile App.

**Reply by brianlovin** [reply]
Got it. This work is being tracked by our team, but I'm not sure exactly when we'll begin working. Thanks for all the extra information!

**Reply by justice-sh** [reply]
Alright. Thanks.

**Reply by dfelton** [reply]
Merely commenting to express that I too would find this helpful. Hopefully this helps feature request gain traction.

The first bullet point would be a great start. Ability to create PR from app.

**Reply by ericges** [reply]
I was looking to create a PR just now too. I too don't care about editing the files, I just want to create a PR from a branch that has recently seen changes.

**Reply by IuliaCMoldovan** [reply]
Same here! For example, you did all the necessary changes, you pushed your commits but you forgot to open a PR and you already closed your laptop and you're on your way... It would be very helpful to just pick a branch and create the PR from the app.

**Reply by JRamirezDD** [reply]
Couple of years and still can't create a PR

**Reply by joacorma** [reply]
Most probably the dev that implemented this feature throughout these 2 years has the latest source ready in his branch but its afk and cannot create the PR from the mobile app. 
It's time to assimilate this could be the case and we will never see this feature released.

**Reply by arnaugomez** [reply]
Please let us create PRs on the iOS app. I hope this comment gets the attention of the devs.

**Reply by karlkaebnick** [reply]
not being able to create PRs is literally my only complaint about the mobile app

**Reply by nhickf** [reply]
I hope this feature to be implemented it's very convenient

**Reply by Tidalikk** [reply]
Still not implemented...

**Reply by Rochdi7** [reply]
Great idea A simplified UI in the GitHub Mobile app for creating pull requests could make mobile contributions much easier for my opinion

**Reply by wallhydj** [reply]
> Thanks for the note @Sherlock-HolmesJM — are you envision that this feature would include file editing and committing from your phone? It would be helpful for us if you could share a few more details about what kind of flow you're in where opening a pull request on mobile would be helpful.
> 
> 
> 
> Thanks!

**Comment by luispurex** [comment]
> Could the feature for creating pull requests on mobile phones be implemented?

**Reply by lkbrian** [reply]
This needs to happen..I may not be around my laptop but always with my phone

**Reply by wallhydj** [reply]
> This needs to happen..I may not be around my laptop but always with my phone

**Comment by cn1t** [comment]
It has been two years since this post and it is still not implemented. I also cant have the github app installed for 2fa and open a pull request in a browser, since it automatically opens the app...

**Reply by aconsuegra** [reply]
Hi @Niteletsplay. Editing code and creating pull requests is already available on Android and iOS. If you have any feedback about the feature, let us know!

**Reply by cn1t** [reply]
Hi @aconsuegra 
Where is that option? I have a branch and want to create a pr for it but cant seem to find a button for that..

**Reply by aconsuegra** [reply]
As I mentioned in [this comment](https://github.com/orgs/community/discussions/6348#discussioncomment-6948884), creating pull requests from existing branches is not possible, yet. But editing files, committing changes, creating a branch and opening a pull request is available.

**Reply by sgolodetz-vm** [reply]
I’ve just been reading through the existing discussion about this. Like others, the thing that would actually be useful to me is to create a pull request from an existing branch. In my case, it would be handy so that I can check through my own changes while I’m not near my PC before getting others to review them. Sometimes I forget to put in a pull request before leaving my PC, and I’d like to do that on mobile.

In terms of feedback on the existing feature: I’m unlikely to want to edit code directly on my phone without at least some kind of external keyboard, so it doesn’t seem that useful to me personally. I’ve already had thumb tendinitis once in life, and that was enough. Plus it’s slow to work that way, annoying to switch between files, and hard to see enough code at once, especially across multiple files.

Conversely, the requested feature has none of those drawbacks, and would be useful out of the box.

**Reply by ikeawesom** [reply]
> it automatically opens the app...

The workaround I have been using is by going into **Incognito Mode** on my mobile browser. This way, when I log into GitHub on my mobile browser, I would not be redirected into the GitHub mobile app. Although it may not be the best solution, I hope it helps anyone!

**Reply by bassmadrigal** [reply]
> Hi @Niteletsplay. Editing code and creating pull requests is already available on Android and iOS. If you have any feedback about the feature, let us know!

Why would you require someone to edit code within the app to submit a pull request from the app?

Why can't we just submit a PR for already edited code that we forgot to submit a PR for on our computer system? This seems like such a no brainer, but we're still here over 3 years later without this highly expected functionality...

**Comment by joeykay9** [comment]
Any updates on this?

**Reply by aconsuegra** [reply]
Hi @joeykay9. Editing code and creating pull requests is already available on Android and iOS. If you have any feedback about the feature, let us know!

**Reply by joeykay9** [reply]
Hi @aconsuegra . Thanks for your prompt response. Is there a doc you can share demonstrating this. Or is this limited to only public repos?

Cos I’ve tried every possible option button in the pull request view and repo overview view and don’t see any feature allowing for the creation of a pull request. Can only view already opened pull requests.

**Reply by cn1t** [reply]
> Hi @aconsuegra . Thanks for your prompt response. Is there a doc you can share demonstrating this. Or is this limited to only public repos?
> 
> Cos I’ve tried every possible option button in the pull request view and repo overview view and don’t see any feature allowing for the creation of a pull request. Can only view already opened pull requests.

Same and btw it doesnt work on public repos either

**Reply by aconsuegra** [reply]
This is a good point, we definitely need better documentation, or improve the feature discoverability.

While you are browsing code in a repository (public or private), you can open a file and tap on the overflow menu icon where you will see an option to "Edit File". Then you will be redirected to a text editor where you can edit the file, and once you are done, you can commit the changes, keep editing more files, or open a pull request.

The functionality to creating files is going to be available pretty soon. And the possibility to create pull requests from existing branches is not planned for now, but we would like to build it at some point.

I hope this helps!

**Reply by joeykay9** [reply]
In my professional opinion, I believe the ability to create pull requests from existing branches is a highly valuable and more critical feature which will require significantly less effort to implement than creating and committing new files; and hence should be prioritised.

**Reply by cn1t** [reply]
Definetily

**Reply by 9glenda** [reply]
I don't think anyone wants to edit code on his phone he rather wants to create a pull request for code he already pushed and tested on his PC.

**Reply by joeykay9** [reply]
> I don't think anyone wants to edit code on his phone he rather wants to create a pull request for code he already pushed and tested on his PC.

Exactly!!

**Reply by smthpy** [reply]
> I don't think anyone wants to edit code on his phone he rather wants to create a pull request for code he already pushed and tested on his PC.

Jokes on you I did, pain.

**Comment by noctyris** [comment]
# YES !

**Reply by we-z** [reply]
How sway

**Comment by lemusalex797** [comment]
Genial

**Comment by notalexandercbooth** [comment]
Test

**Comment by nimeshmaharjan1** [comment]
This is such as a bad ux I need to open a pr from my phone and it keeps redirecting me to the mobile app

**Comment by Vahid8478** [comment]
> > Could the feature for creating pull requests on mobile phones be implemented?
> 
>

**Comment by Vahid8478** [comment]
> It has been two years since this post and it is still not implemented. I also cant have the github app installed for 2fa and open a pull request in a browser, since it automatically opens the app...

**Reply by alirezaraha3** [reply]
Hr__

**Comment by Vahid8478** [comment]
> Any updates on this?

**Reply by alirezaraha3** [reply]
[](https://acrobat.adobe.com/id/urn:aaid:sc:EU:e49118e3-d9d1-4d04-b4ac-0c5e75b345a2)

**Reply by alirezaraha3** [reply]
> [](https://acrobat.adobe.com/id/urn:aaid:sc:EU:e49118e3-d9d1-4d04-b4ac-0c5e75b345a2)

**Comment by Vahid8478** [comment]
> Test

**Reply by alirezaraha3** [reply]
**_@_**

**Comment by Vahid8478** [comment]
> Thanks for the note @Sherlock-HolmesJM — are you envision that this feature would include file editing and committing from your phone? It would be helpful for us if you could share a few more details about what kind of flow you're in where opening a pull request on mobile would be helpful.
> 
> Thanks!

**Comment by April4t** [comment]
Well hopefully a rabbit from my hat

**Comment by AnthonyDugarte** [comment]
Any progress on this? The only reason I open GitHub when out of my PC is to review PRs and create new ones from existing branches.

**Reply by alirezaraha3** [reply]
[](https://acrobat.adobe.com/id/urn:aaid:sc:EU:e49118e3-d9d1-4d04-b4ac-0c5e75b345a2)

**Comment by olawalejuwonm** [comment]
Please any update? I wish I can do this as well

**Comment by ghosty2004** [comment]
Yeah, this need to be implemented...

**Comment by ratheeshks-yeka** [comment]
No Updates Yet 🙄

**Comment by SreevarshAtKoach** [comment]
So I kinda got tired of waiting for this feature so I make minor changes in the app and then raise a PR from my mobile browser .. seems unnecessarily complicated tho

**Reply by aconsuegra** [reply]
Hi @SreevarshPV, thanks for your feedback! Creating pull requets from the GitHub Mobile app is possible after you edit a file and commit your changes. Are you expecting a different behavior?

**Reply by bassmadrigal** [reply]
I think most are looking for the ability to create a PR from an already existing branch that has existing committed changes without requiring someone to edit a file from the app.

For example, you committed changes to a new branch on your computer and left the office, but forgot to submit a PR. I would like to able to open the app, view a branch, and simply submit a PR.

If we look at [this comment](https://github.com/orgs/community/discussions/6348#discussioncomment-1621851) from @brianlovin, I think most are looking for 1, not 2 (which the latter seems to have already been implemented).

**Reply by ericges** [reply]
@aconsuegra I think it is very clear from the comments that people are looking for the opportunity to create PRs from existing branches without editing anything. I don't see why this should be such a difficult feature to implement, as creating PRs from edited files is obviously already a thing. That being said, I think most would never even want to edit a file in the app, let alone create a PR for it. On the other hand, leaving the office and reviewing your own code on your way home and then creating a PR if it looks good is probably a scenario many are familiar with. Please implement this.

**Comment by USBA** [comment]
You might be thinking “Why do you want to create PRs on the mobile phone” and decide not to prioritize developing this feature.

But hear me out. Sometimes I commit changes but forget to create a PR and then I shut down my computer. Currently, I have to restart my computer and wait for it to boot just to create a PR and shut it down again. So it would be extremely convenient if we have the ability to create PRs on the mobile app in scenarios like that.

So please consider developing this feature. Thanks for developing this awesome app. This truly is one of the best iOS apps ever.

**Comment by danirzv** [comment]
This is such a simple feature to have event doesn't need any special UX or any complicated service, why is this feature not present yet

**Comment by tiotamara** [comment]
I'm still waiting for this feature

**Reply by DevinAPorro** [reply]
hello darkness my old friend....

**Comment by nblogist** [comment]
How is this still not a thing!!?

**Reply by wshinwari93** [reply]
@aconsuegra

**Reply by artursmartinaks** [reply]
F

**Comment by ikeawesom** [comment]
The workaround I have been using is by going into **Incognito Mode** on my mobile browser. This way, when I log into GitHub on my mobile browser, I would not be redirected into the GitHub mobile app. Although it may not be the best solution, I hope it helps anyone!

**Reply by marmotking** [reply]
Ask the copilot how to create a PR from the mobile app. It will give you detailed incorrect instructions. Perfect example of precise but not accurate. :)

**Reply by SergVee** [reply]
Thank you for the workaround! It does the trick, I was getting crazy since I followed copilot made up instructions

**Reply by JAWEEWAJ** [reply]
They should do this

**Comment by itsleviaa10** [comment]
hi

**Comment by KevinRJF** [comment]
I like the idea

**Comment by mazenMo77amed** [comment]
I think this a good idea

**Comment by douniwan5788** [comment]
I can't believe you can't even make a PR in the GitHub app

**Reply by nblogist** [reply]
IKR!!

**Comment by jhetuts** [comment]
Up

**Reply by JRamirezDD** [reply]
> Hi

Hi!

**Reply by JRamirezDD** [reply]
Just curious about what kind of scam this thing's running

**Comment by Keinta15** [comment]
Please consider adding this, it would be helpful

**Comment by l14zj** [comment]
‏_xcciu

**Comment by BabyYoda01** [comment]
> It has been two years since this post and it is still not implemented. I also cant have the github app installed for 2fa and open a pull request in a browser, since it automatically opens the app...

**Comment by BabyYoda01** [comment]
> It has been two years since this post and it is still not implemented. I also cant have the github app installed for 2fa and open a pull request in a browser, since it automatically opens the app...

**Comment by Babababababybella** [comment]
Please do add the feature. There are countless people, including myself, who think it would be wonderful and convenient addition!

**Comment by Babababababybella** [comment]
Hey, thanks for writing me! I really need to ask however, with all due respect, what exactly did I do wrong?
I simply handed in a suggestion for the further development of the github app on the phone and voiced agreement that it would be a good addition. Here is the comment in question, standing alone. Not a reply, with no tags or pings.
I look forward to hearing from you because I'm absolutely stumped what the issue is.

Regards,

Baba[Image]


________________________________

**Reply by queenofcorgis** [reply]
Hi there @Babababababybella - this was commented on a comment in the thread to another community member. Thank you for the feedback, I do understand how the notification could be confusing as it was not directed to you. Our team will review our messaging to create more clarity in the community.

**Comment by mrprkr** [comment]
Bump +reopen after merge

**Comment by artursmartinaks** [comment]
🤯

**Comment by aconsuegra** [comment, marked as answer]
Hi everyone! We have added the possibility to create pull requests from existing branches on GitHub Mobile. It's available in version 1.193 and you can read more about the announcement in https://github.blog/changelog/2025-01-22-create-pull-request-from-an-existing-branch-on-github-mobile/ 

Let us know if you have any feedback!

**Reply by Rochdi7** [reply]
sounds Good!!!

**Reply by justice-sh** [reply]
Awesome! Will check and give feedback.

**Reply by justice-sh** [reply]
It works. Thank you for adding this. 🙏

**Comment by aflacake** [comment]
Forcefully, have to do this on the website
